### ISSUE TITLE
[HOLD for payment 2022-09-16] [$250] The file extension shows up twice in the file name reported by @Tushu17

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#10579]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open website/mWeb.
2. Go to chat > Send a jpeg/pdf attachment..
3. See the name of attachment in preview.

## Expected Result:
Extension shouldn't get repeated.

## Actual Result:
It shows up twice.

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- Mobile Web

**Version Number:** 1.1.91-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
<img width="960" alt="2" src="https://user-images.githubusercontent.com/43996225/186734329-f2ed21c2-11ee-4419-a49d-8d86f28cad6f.png">
<img src="https://user-images.githubusercontent.com/43996225/186734425-eed79c17-4ba8-41e0-ab2f-d8bb7f1f0ce7.jpg" width="40%" height="40%">
**Expensify/Expensify Issue URL:** 
**Issue reported by:** @Tushu17 
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by theTrozen77** [comment]
Hi, I have been able to reproduce this and fix it successfully.
`        return {fileName: splitFileName[0], fileExtension};
`        
The filename returned was not correct previously.
Thankyou

**Comment by Puneet-here** [comment]
### Proposal

There are two extensions because we aren't removing the extension from file name and we are adding the extension here too
https://github.com/Expensify/App/blob/d5b1c8249bd3538051abe7d7bc728eaa1e64fd3c/src/components/AttachmentModal.js#L241

The above proposal doesn't work great when a file name has periods in it. It will take the text before the period so the file name will be incomplete. 
We can use `slice(0, -1).join('.')` after split. This way we will get the whole file name without extension then we just have to use that instead of the one with the extension.

**Comment by sonialiap** [comment]
Triaging, can confirm described behavior  
<img width="342" alt="image" src="https://user-images.githubusercontent.com/17131195/187305124-243a20ac-dc63-45f6-a49f-f2016888315d.png">

**Comment by johnmlee101** [comment]
Opening for external, seems like a very easy fix based off of the two proposals already

**Comment by theTrozen77** [comment]
> ### Proposal
> There are two extensions because we aren't removing the extension from file name and we are adding the extension here too
> 
> https://github.com/Expensify/App/blob/d5b1c8249bd3538051abe7d7bc728eaa1e64fd3c/src/components/AttachmentModal.js#L241
> 
> The above proposal doesn't work great when a file name has periods in it. It will take the text before the period so the file name will be incomplete. We can use `slice(0, -1).join('.')` after split. This way we will get the whole file name without extension then we just have to use that instead of the one with the extension.

I believe when Puneet is suggesting here might be an extension to this issue and could be a new ticket. However, my proposal above completely fixes the actual issue here which is that the extension are being repeated twice.

**Comment by Puneet-here** [comment]
It's not an extension, your proposal fixes the issue but breaks the file name when there are periods in it, if a solution breaks something then it's not a good solution

**Comment by sobitneupane** [comment]
**Proposal**
**Reason:**
`splitExtensionFromFileName()` function returns fileName as it is without removing the extension.
https://github.com/Expensify/App/blob/479ca44d7fc9ab55a31c92b82d42eacf83e71f03/src/components/AttachmentModal.js#L115-L120
**Solution:**
As `fileExtension` is already popped from `splitFileName`, we can join `splitFileName` with '.' to get fileName without extension.
```diff
    splitExtensionFromFileName(fullFileName) {
        const fileName = fullFileName.trim();
        const splitFileName = fileName.split('.');
        const fileExtension = splitFileName.pop();
-      return {fileName, fileExtension};
+        return {fileName: splitFileName.join('.'), fileExtension};
    }
```
**Output:**
<img width="1440" alt="Screen Shot 2022-08-31 at 17 02 24" src="https://user-images.githubusercontent.com/25876548/187666623-b45ecffc-a3be-47d7-8b13-27951b96b85f.png">

**Comment by varshamb** [comment]
### Proposal

In `AttachementModal`, `splitExtensionFromFileName()` returns file name with extension.

We should return this instead -
`const fileNameWithoutExtension = fileName.substring(0, fileName.lastIndexOf('.')) || fileName;`

**I think this code more readable.**

https://github.com/Expensify/App/blob/479ca44d7fc9ab55a31c92b82d42eacf83e71f03/src/components/AttachmentModal.js#L115-L120

**Comment by arielgreen** [comment]
https://www.upwork.com/jobs/~01c7f6606ce3547f5f

**Comment by Luke9389** [comment]
I agree that we should go with @sobitneupane's proposal. Go ahead and create a PR @sobitneupane. 
Thank you @sobitneupane, @Puneet-here, @varshamb, @theTrozen77 for your proposals!

@Puneet-here your proposal would work, but I like that @sobitneupane's fixes the problem at the source (the `splitExtensionFromFileName` function).

@theTrozen77 & @Puneet-here, please try to refrain from openly criticizing each others' proposals. If a proposal is chosen and you disagree with the result, that's the time to talk about the flaws you see.

**Comment by Puneet-here** [comment]
> please try to refrain from openly criticizing each others' proposals. If a proposal is chosen and you disagree with the result, that's the time to talk about the flaws you see.

Cool, will remember next time.

**Comment by arielgreen** [comment]
@sobitneupane @Tushu17 @parasharrajat Please check your Upwork inboxes!

**Comment by Luke9389** [comment]
just waiting on payment to close this out melvin.

**Comment by arielgreen** [comment]
thanks @parasharrajat resent

**Comment by arielgreen** [comment]
Paid.
